profiles/common/modules/features/nexteuropa_communities/nexteuropa_communities.module

```diff
@@ -1,7 +1,467 @@
+    'page arguments' => array('nexteuropa_communities_configure_form'),
+    'access arguments' => array('administer group'),
+  );
+  $items['admin/config/nexteuropa_communities/nexteuropa_private_area'] = array(
+    'title' => 'NextEuropa private area',
+    'description' => 'Configure the NextEuropa private area',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('nexteuropa_communities_private_area_form'),
+    'access arguments' => array('administer group'),
+  );
+  return $items;
+}
+
+/**
+ * Manage the configuration form for NextEuropa Communities.
+ */
+function nexteuropa_communities_configure_form() {
+  $form['nexteuropa_communities_url_keyword'] = [
+    '#type' => 'textfield',
+    '#title' => 'Url Keyword',
+    '#description' => 'The url keyword used for community detection',
+    '#default_value' => variable_get('nexteuropa_communities_url_keyword', 'community'),
+  ];
+
+  $options = og_get_all_group_bundle();
+
+  $form['nexteuropa_communities_content_types'] = [
+    '#title' => 'Community group  content types',
+    '#description' => 'Select the content types that are used as a community group',
+    '#type' => 'select',
+    '#multiple' => TRUE,
+    '#required' => TRUE,
+    '#options' => $options,
+    '#default_value' => variable_get('nexteuropa_communities_content_types', array('community')),
+  ];
+
+  return system_settings_form($form);
+}
+
+/**
+ * Provide a form for the private area setting.
+ */
+function nexteuropa_communities_private_area_form() {
+  $form['nexteuropa_communities_private_area'] = [
+    '#type' => 'checkbox',
+    '#title' => 'Enable the private area management for the community part.',
+    '#default_value' => variable_get('nexteuropa_communities_private_area', FALSE),
+  ];
+
+  return system_settings_form($form);
+}
+
+/**
+ * Add a new context negotiation to manage the NextEuropa communities.
+ *
+ * Implements hook_og_context_negotiation_info().
+ */
+function nexteuropa_communities_og_context_negotiation_info() {
+  $providers = [];
+
+  $providers['nexteuropa_communities'] = [
+    'name' => t('NextEuropa URL'),
+    'description' => t("Select group context for any url that start with the configured word, defaults at 'community'"),
+    'callback' => 'nexteuropa_communities_context_handler',
+  ];
+
+  return $providers;
+}
+
+/**
+ * Context handler, get the context from the URL.
+ */
+function nexteuropa_communities_context_handler() {
+  global $language;
+
+  $keyword = variable_get('nexteuropa_communities_url_keyword', 'community');
+
+  // Use the NextEuropa language split for the url to remove language suffix.
+  $path = nexteuropa_multilingual_language_negotiation_split_suffix(request_path(), [$language]);
+  $path_args = explode('/', $path[1]);
+
+  // Look for the community keyword in the path.
+  if ($path_args[0] === $keyword) {
+    $community_path = drupal_lookup_path('source', $path_args[0] . '/' . $path_args[1], $language->language);
```

Comment:
**drupal_lookup_path** : Given an alias, return its Drupal system URL if one exists. Given a Drupal system URL return one of its aliases if such a one exists.
**menu_get_object**: Gets a loaded object from a router item : Gets a loaded object from a router item.

In this case, we want the alias of the current node, drupal_lookup_path is adequate. 

> most probably faster as well
Are you sure ? 
menu_get_object loads object from a menu router: it call access callback function, call load_function and to_arg_functions if exist, build the object so it also call callback function, call all hook_menu_alter, call _menu_translate and  _menu_check_access and menu_get_ancestors etc... 
drupal_lookup_path check and load alias/source from url_alias table, it call only drupal_path_alias_whitelist_rebuild, and it's only contain php code and 1 or 2 database query ...

https://api.drupal.org/api/drupal/includes%21menu.inc/function/menu_get_object/7.x
https://api.drupal.org/api/drupal/includes%21path.inc/function/drupal_lookup_path/7.x